addons/product/models/product_template.py

```diff
@@ -538,7 +538,37 @@ def copy(self, default=None):
             default = {}
         if 'name' not in default:
             default['name'] = _("%s (copy)", self.name)
-        return super(ProductTemplate, self).copy(default=default)
+
+        res = super(ProductTemplate, self).copy(default=default)
+
+        # Since we don't copy the attribute line directly, we need to match the extra price and
+        # exclusion of the newly created attribute line values with the old ones.
+        # This matching expects this extra price/exclusion to be uniquely determined by the attribute value e.g. if
+        # two lines uses the same attribute value, the copied extra price will be the same on both lines.
+
+        # price_extra match
+        template_value_by_value = {
+            template_value.product_attribute_value_id: template_value for template_value in
+            self.product_variant_ids.product_template_attribute_value_ids
+        }
+        for template_value in res.product_variant_ids.product_template_attribute_value_ids:
+            match = template_value_by_value.get(template_value.product_attribute_value_id, False)
+            template_value.price_extra = match.price_extra if match else 0
```

Comment:
```suggestion
        copied_ptavs = res.attribute_line_ids.product_template_value_ids
        for ptav in self.attribute_line_ids.product_template_value_ids:
            if ptav.price_extra:
                copied_ptavs.filtered(
                    lambda x: x.product_attribute_value_id == ptav.product_attribute_value_id
                ).price_extra = ptav.price_extra
```
This is easier to read IMHO